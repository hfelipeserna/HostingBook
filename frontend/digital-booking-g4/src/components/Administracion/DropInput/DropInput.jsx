import React, { useState, useRef, useEffect } from "react";
import DropDownRoom from "../DropDownRoom/DropDownRoom";
import useClickOutside from "../../../hooks/useOnClickOutside";
import useFetch from "../../../hooks/useFetch.js";
import styles from "./DropInput.module.css";

export default function DropInput({ setOnChangeItem, onChangeItem, label, pathname, placeholder, id, dataId}) {
  const [list, setList] = useState(null);
  const [inputContent, setInputContent] = useState("");
  const [dropItems, setDropItems] = useState([
    "Principal",
    "Habitación doble",
    "Habitación simple",
    "Baño",
    "Jacuzzi",
    "Cocina",
    "Hall de entrada",
    "Balcón",
    "Piscina",
    "Churrasquera",
    "Cochera",
    "Patio",
    "Sala de estar",
    "Comedor",
    "Recepción",
    "Ambientes compartidos"
  ]);
  const wrapperRef = useRef(null);
  //const data = useFetch(pathname);
  useClickOutside(wrapperRef, () => setList(null));

  /* useEffect(() => {
    if (data.isLoaded) {
      setDropItems(data.items);
    }
  }, [data.isLoaded, data.items]); */


  const lister = (input) => {
    if (dropItems.length > 0) {
      setList(
        <DropDownRoom
          rooms={dropItems}
          input={input}
          setList={setList}
          setOnChangeItem={setOnChangeItem}
        />
      );
    } else {
      setList(null);
    }
  };

  return (
    <div className={styles.containerDropInput}>
      <label className={styles.labelDropInput}>{label}</label>
      <div className={`${styles.cityContainer} ${styles.divDrawer}`} ref={wrapperRef}>
        {inputContent !== "" ? <div className={`${styles.divDrawer}`}>{list}</div> : ""}
        <input
          type="text"
          className={`${styles.input}`}
          onKeyUp={(e) => {
            lister(e);
            setInputContent(e.target.value);
          }}
          onChange={(e) => {
            setOnChangeItem(e.target.value)
          }}
          value={onChangeItem}
          placeholder={placeholder}
          id={id}
          dataId={dataId}
        />
      </div>
    </div >
  );
}
