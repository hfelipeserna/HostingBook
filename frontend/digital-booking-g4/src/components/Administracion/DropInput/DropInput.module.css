.containerDropInput{
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 0;
}

.labelDropInput{
  color: var(--color-2);
  font-weight: 700;
  margin-top: 20px;
  cursor: pointer;
}

.cityContainer {
  position: relative;
}

.divDrawer{
  display: block;
  gap: 0;
}

.input{
  background-color: var(--color-5);
  border: 3px solid var(--color-5);
  box-shadow: 0px 1px 5px rgb(0 0 0 / 15%);
  color: var(--color-2);
  font-size: 1rem;
  padding: 1vh 40px 1vh 15px;
  width: 100%;
  margin-top: 5px;
}


@media (max-width: 768px) {
  /* .cityContainer {
    min-width: 18.301610541727673vw;
  } */

}

@media (max-width: 480px) {
  .labelDropInput{
    font-size: .9rem;
    margin-top: 7px;
  }

  .input {
    /* position: relative; */
    width: 100%;
    padding: 0.8vh 40px 0.8vh 15px;
  }

  .applyContainer {
    margin: 10px;
  }

}
