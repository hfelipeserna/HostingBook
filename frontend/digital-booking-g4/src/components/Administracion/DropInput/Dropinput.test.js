import DropInput from "./DropInput"
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";


configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Drop input Room test", function () {
    let onChangeItem = ""
    const setOnchangeItem = (palabra) => {
        onChangeItem = palabra
    }

    test("Renderiza Componente Por default", () => {

        const renderizacionSimple = render(
            <DropInput></DropInput>
        );
        expect(renderizacionSimple).toMatchSnapshot()
    });
    test("Testea Cambio de input.value", () => {
        const renderizacionSimple = mount(
            <DropInput setOnChangeItem={setOnchangeItem} onChangeItem={onChangeItem}></DropInput>
        );
        const input = renderizacionSimple.find('input')
        input.simulate('focus');
        input.simulate('change', { target: { value: 'a' } })
        input.simulate('keyDown', {
            which: 27,
            target: {
                blur() {
                    input.simulate('blur');
                },
            },
        });
        expect(renderizacionSimple).toMatchSnapshot()
        expect(onChangeItem).toEqual("a")
    });
});

