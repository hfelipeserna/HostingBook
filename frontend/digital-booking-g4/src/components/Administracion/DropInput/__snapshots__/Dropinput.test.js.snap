// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop input Room test Renderiza Componente Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="containerDropInput"
      >
        <label
          class="labelDropInput"
        />
        <div
          class="cityContainer divDrawer"
        >
          
          <input
            autocomplete="off"
            class="input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="containerDropInput"
    >
      <label
        class="labelDropInput"
      />
      <div
        class="cityContainer divDrawer"
      >
        
        <input
          autocomplete="off"
          class="input"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Drop input Room test Testea Cambio de input.value 1`] = `
<div
  className="containerDropInput"
>
  <label
    className="labelDropInput"
  />
  <div
    className="cityContainer divDrawer"
  >
    <input
      autoComplete="off"
      className="input"
      onChange={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;
